src/device/mod.rs

```diff
@@ -1376,7 +1389,14 @@ impl Runtime {
         }
 
         self.requested_state.exit_node = Some(exit_node);
-        wg_controller::consolidate_wg_state(&self.requested_state, &self.entities).await
+        wg_controller::consolidate_wg_state(&self.requested_state, &self.entities).await?;
+
+        self.entities
```

Comment:
In this case we can probably attempt reseting existing connections only in reconnection state. 

Which would mean, that we should take `self.requested_state.exit_node` before assignment in line 1391, and only if it is `Some(...)` do the call to reset_existing_connections.

Reply:
I agree with checking if there's an old node. However, the flag check should be internal to the adapter itself IMHO.

Reply:
I've added the check if there is a current exit node or if there is a `last_exit_node` set up in case `disconnect_exit_node()` was called beforehand.
However, I've omitted the feature flag check